### ISSUE TITLE
Is there a mistake in the paper about loss function?

### ISSUE THREAD

**Issue by changym3** [pcy1302/DMGI#2]
Hi, thanks for you great work here!
I read the paper Deep Graph Infomax, DGI introduce a discriminator $D$ to maximize the mutual information. (Eq(1) in the paper). And it say this term should be maximized.
But in your final loss function, there is no minus sign in front of it.

I haven't understood DGI much, so I wonder if I got it wrong or there is a written mistake?

Regards,

**Comment by pcy1302** [comment]
Thanks for your interest in our work.
To begin with, the discriminator loss not only maximizes the mutual information between the summary vector and the original graph, but also minimizes it between the summary vector and the corrupted graph.
If you look at the line 112 in DMGI.py,
````logit = self.disc(c, h_1, h_2, samp_bias1, samp_bias2)````
The above line computes D for the original graph and the corrupted graph.

Later in line 39,
````lbl = torch.cat((lbl_1, lbl_2), 1).to(self.args.device)````
The above line concatenates 1s and 0s to generate the labels for the ````logit````.
Then in the end, cross entropy loss is computed between ````logit```` and ````lbl````, which simultaneously maximizes and minimizes the scores returned by the discriminator.

Hopefully this answers your question.

**Comment by changym3** [comment]
Thanks for your prompt reply.
I will think about it further, it helps a lot!
